src/som/interpreter/nodes/ExceptionSignalingNode.java

```diff
@@ -0,0 +1,51 @@
+package som.interpreter.nodes;
+
+import com.oracle.truffle.api.nodes.Node;
+import com.oracle.truffle.api.source.SourceSection;
+
+import bd.primitives.nodes.PreevaluatedExpression;
+import som.vm.Symbols;
+import som.vm.constants.KernelObj;
+import som.vmobjects.SClass;
+import som.vmobjects.SObject;
+
+
+public class ExceptionSignalingNode extends Node {
+
+  @Child protected ExpressionNode getExceptionClassNode;
+  @Child protected ExpressionNode signalExceptionNode;
+  protected final SObject         module;
+
+  public ExceptionSignalingNode(final String exceptionClassName,
```

Comment:
Hm, is that so common that it is worthwhile to have the standard parameter `KernelObj.kernel`?

Reply:
There's a dozen senders, but we can remove it if you want to.

Reply:
How about a named factory method?
I'd prefer that over a constructor, because it would be more explicit.

So, perhaps `createForKernel(..)`?

Reply:
Moved to a factory as requested.